src/actor/Actor.ts

```diff
@@ -42,6 +42,122 @@ declare global {
  */
 export type ValidActorMessageName = keyof ActorMessageType;
 
+export interface actorMixin<T> {
+  actorName?: ValidActorMessageName;
+  onMessage(message: T): void;
+}
+
+interface Constructable<T = {}> {
+  new (...args: any[]): T;
+  prototype: T;
+}
+
+/**
+ * A mixin function to define an Actor type. It creates a class with a stub
+ * for the {@link Actor#onMessage} callback, which must be overwritten.
+ *
+ *    const Actor = actorMixin<MessageType>(SuperClassConstructor);
+ *    class MyActor extends Actor {
+ *      onMessage(message: MessageType) {
+ *        console.log(`Actor ${this.actorName} I received message: ${message}`);
+ *      }
+ *    }
+ *
+ * If you would like to perform some initialization logic, implement the
+ * optional {@link Actor#init} callback.
+ *
+ *    const Actor = actorMixin<MessageType>(SuperClassConstructor);
+ *    class MyActor extends Actor {
+ *      stockData?: StockData;
+ *      count?: number;
+ *
+ *      async init() {
+ *        this.count = 0;
+ *        this.stockData = await (await fetch("/stockdata.json")).json()
+ *      }
+ *
+ *      onMessage(message: MessageType) {
+ *        this.count!++;
+ *        console.log(`Actor ${this.actorName} received message number ${this.count}: ${message}`);
+ *      }
+ *    }
+ *
+ * If you want to know the actorName that this actor is assigned to in your
+ * application, you can use `actorName`. This field is accessible only after
+ * the {@link hookup} has been called.
+ *
+ * Users of this mixin generally should not use {@link Actor#initPromise}. This
+ * is an internal implementation detail for {@link hookup}.
+ */
+export function actorMixin<T, S extends Constructable = Constructable<Object>>(
+  superClass: S
+) {
+  class actor extends superClass {
```

Comment:
Nit: can be `return class extends superClass`

Reply:
Done.